src/node_api.cc

```diff
@@ -557,65 +576,67 @@ void napi_module_register(napi_module* mod) {
   node::node_module_register(nm);
 }
 
-#define RETURN_STATUS_IF_FALSE(condition, status)                       \
+#define RETURN_STATUS_IF_FALSE(env, condition, status)                  \
   do {                                                                  \
     if (!(condition)) {                                                 \
-      return napi_set_last_error((status));                             \
+      return napi_set_last_error((env), (status));                      \
     }                                                                   \
   } while (0)
 
-#define CHECK_ARG(arg) RETURN_STATUS_IF_FALSE((arg), napi_invalid_arg)
+#define CHECK_ENV(env) \
+  assert((env) != nullptr)
```

Comment:
I’d say it’s nicer than segfaulting, and when it’s hit it’s always a sign that the program is in an invalid state… that sounds good enough for aborting to me?